lib/composer-cmds.ts

```diff
@@ -1,11 +1,32 @@
 import * as path from 'path';
 import * as childProcess from 'child_process';
 
-export const composerCmd = {command: 'composer', args: []};
-export const composerPharCmd = {command: 'php', args: [`${path.resolve(path.resolve() + '/composer.phar')}`]};
+class Command {
+  protected constructor(readonly command: string, readonly args: string[]) {
+  }
+
+  protected withAdditionalArgs(args: string[]): Command {
+    return new Command(this.command, [...this.args, ...args]);
+  }
+}
+
+export class Composer extends Command {
```

Comment:
I'm not convinced if this class & inheritance hierarchy with static functions is easier to follow than the procedural code that existed before.

Perhaps I'm missing something, but could we instead export some functions for "version" and "listPlatformDeps" instead of this inheritance hierarchy and class-based approach?

Reply:
done